core/ssh_keys_test.go

```diff
@@ -0,0 +1,453 @@
+package core
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"testing"
+
+	"github.com/krystal/go-katapult"
+	"github.com/krystal/go-katapult/internal/test"
+	"github.com/krystal/go-katapult/internal/testclient"
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_AuthSSHKey_JSONMarshaling(t *testing.T) {
+	tests := []struct {
+		name string
+		obj  *AuthSSHKey
+	}{
+		{
+			name: "empty",
+			obj:  &AuthSSHKey{},
+		},
+		{
+			name: "full",
+			obj: &AuthSSHKey{
+				ID:          "a",
+				Name:        "b",
+				Fingerprint: "c",
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			testJSONMarshaling(t, tt.obj)
+		})
+	}
+}
+
+func Test_sshKeyResponseBody_JSONMarshaling(t *testing.T) {
+	tests := []struct {
+		name string
+		obj  *sshKeysResponseBody
+	}{
+		{
+			name: "empty",
+			obj:  &sshKeysResponseBody{},
+		},
+		{
+			name: "ssh_key",
+			obj: &sshKeysResponseBody{
+				SSHKey: &AuthSSHKey{},
+			},
+		},
+		{
+			name: "ssh_keys",
+			obj: &sshKeysResponseBody{
+				SSHKeys: []*AuthSSHKey{{}},
+			},
+		},
```

Comment:
We need to add the `Pagination` field to this test, to ensure we can keep serializing it correctly.

However, as we only really care about JSON serialization structure here, I think we can just squash these test cases into a single `"full"` test case which sets all fields on the struct to non-zero values:

```suggestion
		{
			name: "full",
			obj: &sshKeysResponseBody{
				Pagination: &katapult.Pagination{
					CurrentPage: 33,
				},
				SSHKey:  &AuthSSHKey{ID: "1"},
				SSHKeys: []*AuthSSHKey{{ID: "2"}},
			},
		},
```